lib/charms/layer/basic.py

```diff
@@ -157,7 +158,15 @@ def apt_install(packages):
            '--option=Dpkg::Options::=--force-confold',
            '--assume-yes',
            'install']
-    check_call(cmd + packages, env=env)
+    for attempt in range(3):
+        try:
+            check_call(cmd + packages, env=env)
+        except CalledProcessError:
+            if attempt == 2:  # third attempt
+                raise
+            sleep(5)
+        else:
```

Comment:
I answered this via other channels, but for posterity:

The `else` here is on the `try` / `except` block, and will be executed if there are no exceptions raised.  It is almost the same thing as putting the `break` inside the `try` block except that exceptions raised in the `else` block are not caught by any `except` blocks (though in this case, `break` can't raise a `CalledProcessError` anyway).